src/test/java/api/loans/CheckOutByBarcodeTests.java

```diff
@@ -857,9 +858,36 @@ public void canCheckOutInProcessItemWithRequest() {
     assertThat(smallAngryPlanet, hasItemStatus(CHECKED_OUT));
   }
 
+  @Test
+  public void cannotCheckOutWhenItemIsNotLoanable() {
+    IndividualResource notLoanablePolicy = loanPoliciesFixture.create(
+      new LoanPolicyBuilder()
+        .withName("Not Loanable Policy")
+        .withLoanable(false));
+
+    useFallbackPolicies(
+      notLoanablePolicy.getId(),
+      requestPoliciesFixture.allowAllRequestPolicy().getId(),
+      noticePoliciesFixture.inactiveNotice().getId(),
+      overdueFinePoliciesFixture.facultyStandard().getId(),
+      lostItemFeePoliciesFixture.facultyStandard().getId());
+
+    InventoryItemResource nod = itemsFixture.basedUponNod();
+    IndividualResource steve = usersFixture.steve();
+    Response response = loansFixture.attemptCheckOutByBarcode(nod, steve);
+
+    assertThat(response.getJson(), hasErrorWith(allOf(
+      hasMessage("Item is not loanable"),
+      hasItemBarcodeParameter(nod),
+      hasLoanPolicyParameters(notLoanablePolicy))));
+  }
+
   @Test
   public void checkOutFailsWhenCirculationRulesReferenceInvalidLoanPolicyId() {
-    setInvalidLoanPolicyReferenceInRules("some-loan-policy");
+    UUID invalidLoanPolicyId = UUID.randomUUID();
+    loanPoliciesFixture.create(invalidLoanPolicyId);
+    setInvalidLoanPolicyReferenceInRules(invalidLoanPolicyId.toString());
```

Comment:
What does `setInvalidLoanPolicyReferenceInRules` do?

Reply:
It updates circulation rules with unknown id of loan policy and fixed ids for the others.

Reply:
As I understand it, this updates the rules with the ID of a loan policy that does exist at the time. Why is there a need to have a special method for a successful operation?

Reply:
When we try to save the rule with new ID, the validation will not be passed, because the fixture doesn't have this loan policy ID. I added it to the fixture before rule saving and removed it after.

Reply:
I think I might be missing something. Is the sequence something like below?

# Create example loan policy
# Update circulation rules to refer to example loan policy
# Delete example loan policy
# Attempt to check out an item

Reply:
Yes, that's right. Actually I just added the policy to the fixture and removed it after circ rules update, to avoid validation error. The rest of the test had been implemented before.

Reply:
Ok, in which case, at the point when the rules are updated, the loan policy reference is ok. I think that means that normal methods for updating the rules can be used and there doesn't need to be a new one?

Reply:
Sorry, I didn't understand.
I didn't create setInvalidLoanPolicyReferenceInRules method, it had been created before.
Do you mean that we shouldn't use it and call updateCirculationRules() directly instead of calling setInvalidLoanPolicyReferenceInRules()?

circulationRulesFixture.updateCirculationRules(
      circulationRulesFixture.soleFallbackPolicyRule(invalidLoanPolicyReference,
        requestPoliciesFixture.allowAllRequestPolicy().getId().toString(),
        noticePoliciesFixture.inactiveNotice().getId().toString(),
        overdueFinePoliciesFixture.facultyStandard().getId().toString(),
        lostItemFeePoliciesFixture.facultyStandard().getId().toString()));

Reply:
I didn't realise that wasn't introduced by you. Yeah, I was wondering why the regular update wasn't used. Let's leave this for now.